### ISSUE TITLE
Change the `result` type to match the one proposed for pervasives.

### ISSUE THREAD

**Issue by atavener** [dbuenzli/tsdl#21]
Also to match the Rresult module!

**Comment by dbuenzli** [comment]
Yes. We should simply make it depend on the `result` compatibility package and redefine 
```ocaml
type 'a result = [ `Error of string | `Ok of 'a ] 
```
in `Tsdl.Sdl` to 
```ocaml
type 'a result = ('a, [`Msg of string]) Result.t
```

**Comment by Octachron** [comment]
I might have the time to implement this change, I just have one question:
why add an intermediary constructor `Msg` rather than use directly 
```OCaml
type 'a result = ( 'a, string ) Result.result
```
?

**Comment by dbuenzli** [comment]
It helps with composing with other libraries that may expose richer error messages.

**Comment by Octachron** [comment]
Ok, thanks for the explanation.

**Comment by dbuenzli** [comment]
This was done in 3c5335a7c359435d1efc4725 and released in v0.9.0. Thanks to @Octachron for the patch.
